Controller/ReceivedUrl.php

```diff
@@ -0,0 +1,250 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2018 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Controller;
+
+use Bolt\Boltpay\Helper\Log as LogHelper;
+use Bolt\Boltpay\Helper\Cart as CartHelper;
+use Bolt\Boltpay\Helper\Config as ConfigHelper;
+use Bolt\Boltpay\Helper\Bugsnag;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Checkout\Model\Session as CheckoutSession;
+use Magento\Quote\Model\Quote;
+use Magento\Sales\Model\Order;
+use Bolt\Boltpay\Helper\Order as OrderHelper;
+use Bolt\Boltpay\Controller\ReceivedUrl as ReceivedUrlTrait;
+
+trait ReceivedUrl
+{
+    /**
+     * @var LogHelper
+     */
+    private $logHelper;
+    /**
+     * @var ConfigHelper
+     */
+    private $configHelper;
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+    /**
+     * @var CheckoutSession
+     */
+    private $checkoutSession;
+    /**
+     * @var OrderHelper
+     */
+    private $orderHelper;
+
+    /**
+     * @return \Magento\Framework\App\ResponseInterface
+     */
+    public function execute()
+    {
+        $boltSignature = $this->getRequest()->getParam('bolt_signature');
+        $boltPayload = $this->getRequest()->getParam('bolt_payload');
+
+        $signature = base64_decode($boltSignature);
+
+        $magentoSavedSignature = $this->configHelper->getSigningSecret();
```

Comment:
$secret

Reply:
@daisy1754 fixed in separate PR https://github.com/BoltApp/bolt-magento2/pull/299/files